### ISSUE TITLE
VictoriaMetrics Benchmark against M3

### ISSUE THREAD

**Issue by fr3akX** [VictoriaMetrics/VictoriaMetrics#5466]
### Describe the solution you'd like

M3 holds a significant position in the metrics database market. Given that VictoriaMetrics has conducted benchmarks against all major time-series databases, it would be valuable to have a benchmark specifically against M3. This addition would help fill an important gap in the comparative analysis of time-series databases

**Comment by hagen1778** [comment]
Thanks for suggestion @fr3akX! I'll keep it in mind when next time thinking about a blogpost. 
We're also open for a third-party independent tests!

Btw, m3db [latest release](https://github.com/m3db/m3/releases/tag/v1.5.0) was more than a year ago - is it still used? Are there any forks with up-to-date version?

**Comment by naughtyGitCat** [comment]
@hagen1778 yes m3db is less active than vm, and we need a migrate function in `vmctl` to migrate data from `m3db` cluster to `VictorialMetrics`, is there any plan to do this?

since there already exists some migrate function in `vmctl` like `Cortex `, `Mimir`, `Thanos`， why not another function help migrate `m3db` cluster to `VictorialMetrics`?

best wishes

**Comment by hagen1778** [comment]
Hey @naughtyGitCat! Thanks for the suggestion!
We usually make decisions to add something based on community demands. The mentioned migration mods were just more popular, I guess. And another thing is that we don't have engineers familiar with m3db. 
It would be nice to add a migration path from m3db to VictoriaMetrics, and we're open to suggestions and contributions. Do you mind opening a feature request so we can track demand on it?

**Comment by naughtyGitCat** [comment]
@hagen1778 it seems possible to use `vmctl remote-read` to migrate data from `m3db` to `vm`, but need a little trick, because the `remote-read` hardcoded the prefix `remoteReadPath`

after changed it to `remoteReadPath     = "/api/v1/prom/remote/read"` , it works with `stream=false`.

later time , i will pull reqest to make `remoteReadPath` be alterable by add a new argument `--remote-read-path` with default value `/api/v1/read` (previous hardcoded)

**Comment by dtsdwarak** [comment]
Hi @naughtyGitCat 

We're trying to a similar comparison between M3DB and VictoriaMetrics.

If you've made some progress, can you share some references/results?

Thanks in advance!

**Comment by naughtyGitCat** [comment]
> Hi @naughtyGitCat
> 
> We're trying to a similar comparison between M3DB and VictoriaMetrics.
> 
> If you've made some progress, can you share some references/results?
> 
> Thanks in advance!

@dtsdwarak 

We have lots of data, long time range transfer will crash, finally, we cut the time range to a month, and you can also use `--remote-read-filter-label-value` to transfer series by endpoint type.

for example:

```py
monthRange = GetRangeByMonth()

labels = ['mysql', 'node', 'windows', 'redis']

for month in monthRange:
    for labels in labels:
        # do transfer domain/realm series by label and month
        cmd = generate_cmd(month, label)
        os.run(cmd)
```
below is the cmd example

```sh
 ./vmctl-linux-amd64 remote-read --remote-read=true --remote-read-filter-label-value='^node.*' --remote-read-src-addr=http://M3DB-QUERY-COMPONENT:7202 --remote-read-filter-time-start=2022-04-04T00:00:00+08:00 --remote-read-filter-time-end=2024-04-04T23:59:59+08:00 --remote-read-step-interval=minute --vm-addr=http://127.0.0.1:8428 --vm-concurrency=8  --remote-read-concurrency=8
```


hope this can help u



the m3db developers give up the m3db commercial business and start with a new CI/CD business.

now we write time series both to m3db cluster and VM single nodes, after some months, we will tear down the m3db cluster

**Comment by dtsdwarak** [comment]
thanks a lot @naughtyGitCat
